kafka/src/test/java/com.expedia.adaptivealerting.kafka/KafkaDetectorManagerTest.java

```diff
@@ -110,28 +143,59 @@ private void initConfig() {
+        this.metric_weakAnomaly = TestObjectMother.mappedMetricData();
+        this.metric_strongAnomaly = TestObjectMother.mappedMetricData();
+        this.metric_modelWarmup = TestObjectMother.mappedMetricData();
+        this.metric_unknownAnomaly = TestObjectMother.mappedMetricData();
     }
     
     private void initDependencies() {
-        when(detectorManager.getDetectorTypes())
-                .thenReturn(new HashSet<>(Arrays.asList("constant-detector", "ewma-detector")));
+        when(detectorManager.hasDetectorType(anyString())).thenReturn(true);
+        
+        when(detectorManager.classify(metric_normalAnomaly))
+                .thenReturn(TestObjectMother.anomalyResult(AnomalyLevel.NORMAL));
+        when(detectorManager.classify(metric_weakAnomaly))
+                .thenReturn(TestObjectMother.anomalyResult(AnomalyLevel.WEAK));
+        when(detectorManager.classify(metric_strongAnomaly))
+                .thenReturn(TestObjectMother.anomalyResult(AnomalyLevel.STRONG));
+        when(detectorManager.classify(metric_modelWarmup))
+                .thenReturn(TestObjectMother.anomalyResult(AnomalyLevel.MODEL_WARMUP));
+        when(detectorManager.classify(metric_unknownAnomaly))
+                .thenReturn(TestObjectMother.anomalyResult(AnomalyLevel.UNKNOWN));
     }
     
     private void initTestMachinery() {
         
         // Topology test drivers
         val topology = new KafkaAnomalyDetectorManager(saConfig, detectorManager).buildTopology();
-        this.logAndFailDriver = TestObjectMother.topologyTestDriver(topology, MetricData.class, false);
-        this.logAndContinueDriver = TestObjectMother.topologyTestDriver(topology, MetricData.class, true);
+        this.logAndFailDriver = TestObjectMother.topologyTestDriver(topology, MappedMetricDataJsonSerde.class, false);
+        this.logAndContinueDriver = TestObjectMother.topologyTestDriver(topology, MappedMetricDataJsonSerde.class, true);
         
         // MetricData sources
-        this.mappedMetricDataFactory = TestObjectMother.mappedMetricDataFactory();
         this.stringFactory = TestObjectMother.stringFactory();
+        this.metricFactory = TestObjectMother.mappedMetricDataFactory();
         
         // MappedMetricData consumers
-        this.stringDeserializer = new StringDeserializer();
-        this.anomalyResultDeserializer = TestObjectMother.anomalyResultDeserializer();
+        this.stringDeser = new StringDeserializer();
+        this.anomalyDeser = new MappedMetricDataJsonDeserializer();
+    }
+    
+    private void publishesAnomaly(MappedMetricData metric, AnomalyLevel anomalyLevel) {
+        val anomalyRecord = getAnomalyRecord(metric);
+        val anomaly = anomalyRecord.value();
+        assertNotNull(anomaly);
+        assertEquals(anomalyLevel, anomaly.getAnomalyResult().getAnomalyLevel());
+    }
+    
+    @SuppressWarnings("unused")
+    private void doesNotPublishAnomaly(MappedMetricData metric) {
```

Comment:
Is this for future use or am I missing its purpose?

Reply:
Well, previously I was verifying that we _don't_ publish certain conditions, but after we changed it, I thought I'd keep this logic around in case we wanted to suppress something. Let me know if you prefer that I simply remove it. I can go either way.